spark/src/test/scala/org/apache/spark/sql/delta/optimize/OptimizeCompactionSuite.scala

```diff
@@ -536,6 +536,39 @@ trait OptimizeCompactionSuiteBase extends QueryTest
     }
   }
 
+  test("optimize command with batching") {
```

Comment:
Can you add more tests to cover:
1. `OPTIMIZE WHERE` for patititioned table. This makes sure the batching works correctly with the filter.
2. Since this change also impacts  `zorder by` and `cluster by`, we need to add tests for both of them to validate batching works as expected.
3. OPTIMIZE on an empty table. Make sure it doens't trigger any divide by zero errors.

Reply:
Thanks I'll work on those

Reply:
Added tests for all the above